src/triggers/projectUpdateV2.ts

```diff
@@ -0,0 +1,182 @@
+  };
+}
+
+type EventType = "create" | "update";
+
+const subscribeHook = (eventType: EventType) => async (z: ZObject, bundle: Bundle) => {
+  z.console.log("bundle", bundle);
+  const data = {
+    url: bundle.targetUrl,
+    inputData:
+      bundle.inputData && Object.keys(bundle.inputData).length > 0
+        ? pick(bundle.inputData, ["creatorId", "projectId", "teamId"])
+        : undefined,
+  };
+
+  const webhookType = eventType === "create" ? "createProjectUpdate" : "updateProjectUpdate";
+  return z
+    .request({
+      url: `https://client-api.linear.app/connect/zapier/subscribe/${webhookType}`,
+      method: "POST",
+      body: data,
+    })
+    .then((response) => response.data);
+};
+
+const getProjectUpdateList = () => async (z: ZObject, bundle: Bundle) => {
+  const variables: Record<string, string> = {};
+  const variableSchema: Record<string, string> = {};
+  const filters: unknown[] = [];
+  if (bundle.inputData.creatorId) {
+    variableSchema.creatorId = "ID";
+    variables.creatorId = bundle.inputData.creatorId;
+    filters.push({ user: { id: { eq: new VariableType("creatorId") } } });
+  }
+  if (bundle.inputData.teamId) {
+    variableSchema.teamId = "ID";
+    variables.teamId = bundle.inputData.teamId;
+    filters.push({ project: { accessibleTeams: { id: { eq: new VariableType("teamId") } } } });
+  }
+  if (bundle.inputData.projectId) {
+    variableSchema.projectId = "ID";
+    variables.projectId = bundle.inputData.projectId;
+    filters.push({ project: { id: { eq: new VariableType("projectId") } } });
+  }
+  const filter = { and: filters };
+
+  const jsonQuery = {
+    query: {
+      __variables: variableSchema,
+      projectUpdates: {
+        __args: {
+          first: 25,
+          filter,
+        },
+        nodes: {
+          id: true,
+          body: true,
+          url: true,
+          createdAt: true,
+          updatedAt: true,
+          editedAt: true,
+          health: true,
+          project: {
+            id: true,
+            name: true,
+          },
+          user: {
+            id: true,
+            name: true,
+            email: true,
+          },
+        },
+      },
+    },
+  };
+  const query = jsonToGraphQLQuery(jsonQuery);
+
+  const response = await z.request({
+    url: "https://api.linear.app/graphql",
+    headers: {
+      "Content-Type": "application/json",
+      Accept: "application/json",
+      authorization: bundle.authData.api_key,
+    },
+    body: {
+      query,
+      variables,
+    },
+    method: "POST",
+  });
+
+  const data = (response.json as ProjectUpdatesResponse).data;
+  return data.projectUpdates.nodes;
+};
+
+const operationBase = {
+  inputFields: [
+    {
+      required: false,
+      label: "Creator",
+      key: "creatorId",
+      helpText: "Only trigger on project updates added by this user.",
+      dynamic: "user.id.name",
+      altersDynamicFields: true,
+    },
+    {
+      required: false,
+      label: "Project ID",
```

Comment:
Considered on some of the users – why is this using ID while the rest show the name?

Reply:
This isn't a dynamically loaded field where you pick from a dropdown list of options

Reply:
But is there any reason why it couldn't be?

Reply:
Workspaces typically have way more projects than users and teams. The dropdown is kind of annoying with a long list because you have to load the dropdown page by page, check if your value is present, load a new page of values if not, and keep going until you find what you want.

Reply:
I don't know, there can be thousands of users too – there must be a way to pass the search query through to the API request to avoid the need to scroll through pages.

Lets take a look at this separately across all these triggers?

Reply:
I'll create a separate followup